### ISSUE TITLE
Remove unnecessary Karaf features

### ISSUE THREAD

**Issue by vincent-zurczak** [roboconf/roboconf-platform#372]
Karaf 4 is much more modular than previous versions.
In particular, the core feature was split. Please, remove the ones Roboconf does not need.

**Comment by bourretp** [comment]
After a few tries, I have removed the following Karaf features. Some of them are useful for debug, but can always be added using `feature:install`. The only one I've kept is the `log` feature, which holds a very special place in my heart ;)

- `jaas`
- `diagnostic`
- `service`
- `webconsole`
- `deployer`
- `system`
- `package`
- `management`
- `ipojo-webconsole`

These features has been removed from both the DM and the Agent distributions.

Results

| Archive                                        | Before | After    | Ratio    |
|------------------------------------------|----------|--------   |-----------|
| roboconf-karaf-dist-dm.tar.gz     | 28 Mio  | 27 Mio | -5.6%   |
| roboconf-karaf-dist-dm.zip         | 28 Mio  | 27 Mio | -5.6%   |
| roboconf-karaf-dist-agent.tar.gz | 20 Mio  | 14 Mio | -30%    |
| roboconf-karaf-dist-agent.zip     | 21 Mio  | 15 Mio | -26.8% |

**Comment by vincent-zurczak** [comment]
OK for the agent.
But for the DM, please, keep the web consoles. 1 MB is not enough significant.
Also, did you try to install an extension with your light distribution?

Are you sure connecting with the Karaf client will work if you remove **jaas**?

**Comment by bourretp** [comment]
- Ok, I'll re-add the webconsoles on the DM.
- ??? An extension ??? What the hell is that? A target? or you're speaking OCCI ;)
- For the third point, I don't know if it is possible now, or if it was possible before. I'll ask to an expert:  @gibello

**Comment by vincent-zurczak** [comment]
An extension, for the DM, is a target handler. :smile:

**Comment by bourretp** [comment]
Oh I see.
BTW the `target` command does not seems to work properly (even after a last minute pull). Versions still have a trailing `.0`, e.g. `0.5.0-SNAPSHOT`

**Comment by vincent-zurczak** [comment]
> BTW the target command does not seems to work properly (even after a last minute pull). Versions still have a trailing .0, e.g. 0.5.0-SNAPSHOT

:open_mouth:

**Comment by bourretp** [comment]
See associated pull request #379 

Connection to the karaf using the `client` command does work. There is a warning about unverified keys, but nothing blocking.
```
6919 [sshd-SshClient[4e04a765]-nio2-thread-1] WARN org.apache.sshd.client.keyverifier.AcceptAllServerKeyVerifier - Server at [/0.0.0.0:8101, RSA, 60:22:ba:06:f2:0e:a4:93:2f:37:04:d6:f6:76:7c:0b] presented unverified {} key: {}
```

Installing a target using the karaf command fails:
```
roboconf-dm > target jclouds 
Executing feature:install jclouds-for-roboconf...
Executing bundle:install --start mvn:net.roboconf/roboconf-target-jclouds/0.5.0.SNAPSHOT...
Bundle IDs: 
Error executing command: Error installing bundles:
	Unable to install bundle mvn:net.roboconf/roboconf-target-jclouds/0.5.0.SNAPSHOT
```

But installing it manually, with the right URL works:
```
roboconf-dm > bundle:install -s mvn:net.roboconf/roboconf-target-jclouds/0.5-SNAPSHOT
Bundle ID: 122
```

**Comment by vincent-zurczak** [comment]
Actually, I was more interested by `feature:install jclouds-for-roboconf`.
I think I saw deploying features was part of the **deployer** feature.

And there is indeed a bug with the Roboconf target command. I'm fixing it...

**Comment by bourretp** [comment]
Mmmh, I think it's related to the `feature` feature (*sic*).
Here is the formal proof ;)

```
roboconf-dm > feature:list --installed 
```

|Name                 | Version          | Required | State   | Repository                        | Description|
|---------------------|-------------------|------------|----------|----------------------------------|------------------------------------------|
pax-jetty            | 9.2.10.v20150310 |          | Started | org.ops4j.pax.web-4.1.4           | Provide Jetty engine support
pax-http-jetty       | 4.1.4            |          | Started | org.ops4j.pax.web-4.1.4           |
pax-http             | 4.1.4            |          | Started | org.ops4j.pax.web-4.1.4           | Implementation of the OSGI HTTP Service
pax-http-whiteboard  | 4.1.4            |          | Started | org.ops4j.pax.web-4.1.4           | Provide HTTP Whiteboard pattern support
pax-war              | 4.1.4            |          | Started | org.ops4j.pax.web-4.1.4           | Provide support of a full WebContainer
roboconf-dm          | 0.5.0.SNAPSHOT   | x        | Started | roboconf-agent-karaf-0.5-SNAPSHOT |
jersey-1.x           | 1.18.2           | x        | Started | roboconf-agent-karaf-0.5-SNAPSHOT |
ipojo                | 1.12.1           |          | Started | org.apache.felix.ipojo.features   | Apache Felix iPOJO Core Runtime
ipojo-command        | 1.12.1           |          | Started | org.apache.felix.ipojo.features   | Apache Felix iPOJO Shell Command
ipojo-all            | 1.12.1           | x        | Started | org.apache.felix.ipojo.features   | Apache Felix iPOJO All Runtime Bundles
ipojo-webconsole     | 1.12.1           | x        | Started | org.apache.felix.ipojo.features   | Apache Felix iPOJO Web Console Plugin
aries-proxy          | 4.0.0            |          | Started | standard-4.0.0                    | Aries Proxy
aries-blueprint      | 4.0.0            | x        | Started | standard-4.0.0                    | Aries Blueprint
feature              | 4.0.0            | x        | Started | standard-4.0.0                    | Features Support
shell                | 4.0.0            |          | Started | standard-4.0.0                    | Karaf Shell
shell-compat         | 4.0.0            | x        | Started | standard-4.0.0                    | Karaf Shell Compatibility
bundle               | 4.0.0            | x        | Started | standard-4.0.0                    | Provide Bundle support
config               | 4.0.0            | x        | Started | standard-4.0.0                    | Provide OSGi ConfigAdmin support
instance             | 4.0.0            | x        | Started | standard-4.0.0                    | Provide Instance support
jaas                 | 4.0.0            |          | Started | standard-4.0.0                    | Provide JAAS support
log                  | 4.0.0            | x        | Started | standard-4.0.0                    | Provide Log support
http                 | 4.0.0            |          | Started | standard-4.0.0                    | Implementation of the OSGI HTTP Service
war                  | 4.0.0            | x        | Started | standard-4.0.0                    | Turn Karaf as a full WebContainer
webconsole           | 4.0.0            | x        | Started | standard-4.0.0                    | Base support of the Karaf WebConsole
ssh                  | 4.0.0            | x        | Started | standard-4.0.0                    | Provide a SSHd server on Karaf

Notice that the the deployer feature is absent. The jaas feature is there, which means it is required by another feature we include.

```
roboconf-dm > feature:install jclouds-for-roboconf
roboconf-dm > feature:list --installed 
```

|Name                 | Version          | Required | State   | Repository                        | Description|
|---------------------|-------------------|------------|----------|----------------------------------|------------------------------------------|
… | … | … | … | … |
jclouds-for-roboconf | 0.5.0.SNAPSHOT   | x        | Started | roboconf-agent-karaf-0.5-SNAPSHOT |

```
roboconf-dm > bundle:list
START LEVEL 100 , List Threshold: 50
```


| ID | State    | Lvl | Version        | Name|
|-----|----------|------|----------------|--------------------------------------------------------|
… | … | … | … | … |
107 | Active   |  80 | 2.2.2          | Gson
108 | Active   |  80 | 17.0.0         | Guava: Google Core Libraries for Java
109 | Active   |  80 | 3.0.0          | guice, Fragments: 111, 110, 119
110 | Resolved |  80 | 3.0.0          | guice-assistedinject, Hosts: 109
111 | Resolved |  80 | 3.0.0          | guice-multibindings, Hosts: 109
112 | Active   |  80 | 1.8.1          | jclouds compute core
113 | Active   |  80 | 1.8.1          | jclouds Components Core
114 | Active   |  80 | 1.8.1          | jclouds script builder
115 | Active   |  80 | 1.8.1          | jclouds openstack-keystone api
116 | Active   |  80 | 1.8.1          | jclouds openstack-nova api
117 | Active   |  80 | 1.0.0.5        | Apache ServiceMix :: Bundles :: aopalliance
118 | Active   |  80 | 1.0.0.1        | Apache ServiceMix :: Bundles :: javax.inject
119 | Resolved |  80 | 6.2.0          | 99soft :: Rocoto, Hosts: 109

Yeah, I like tables...
